crates/holochain/tests/tests/countersigning/session_interaction_over_websocket.rs

```diff
@@ -0,0 +1,810 @@
+        response
+    } else {
+        unreachable!();
+    };
+
+    // Countersigning session state should exist for both agents and be in "Accepted" state.
+    assert_matches!(
+        get_session_state(&alice.cell_id, &alice_app_tx).await,
+        Some(CountersigningSessionState::Accepted(_))
+    );
+    assert_matches!(
+        get_session_state(&bob.cell_id, &bob_app_tx).await,
+        Some(CountersigningSessionState::Accepted(_))
+    );
+
+    // Bob commits entry and shuts down.
+    for _ in 0..5 {
+        let response = call_zome_fn_fallible(
+            &bob_app_tx,
+            bob.cell_id.clone(),
+            &bob.signing_keypair,
+            bob.cap_secret,
+            TestWasm::CounterSigning.coordinator_zome_name(),
+            "create_a_countersigned_thing_with_entry_hash".into(),
+            &[alice_response.clone(), bob_response.clone()],
+        )
+        .await;
+
+        if let AppResponse::ZomeCalled(_) = response {
+            break;
+        }
+        tokio::time::sleep(Duration::from_millis(100)).await;
+    }
+
+    tracing::info!("Bob committed countersigned entry - shutting down conductor.\n");
+
+    let bob_config = bob.shutdown();
+
+    // Alice commits countersigned entry.
+    for _ in 0..5 {
+        let response = call_zome_fn_fallible(
+            &alice_app_tx,
+            alice.cell_id.clone(),
+            &alice.signing_keypair,
+            alice.cap_secret,
+            TestWasm::CounterSigning.coordinator_zome_name(),
+            "create_a_countersigned_thing_with_entry_hash".into(),
+            &[alice_response.clone(), bob_response.clone()],
+        )
+        .await;
+
+        if let AppResponse::ZomeCalled(_) = response {
+            break;
+        }
+        tokio::time::sleep(Duration::from_millis(100)).await;
+    }
+
+    tracing::info!("Alice committed countersigned entry - restarting conductor to provoke unresolved state...\n");
+
+    // Restart Alice's conductor to put countersigning session in unresolved state.
+    let mut alice = alice.restart_conductor().await;
+
+    // Attach app interface to Alice's conductor.
+    let (alice_app_tx, mut alice_app_rx) = alice.connect_app_interface().await;
+    // Spawn task listening for signals.
+    tokio::spawn(async move { while alice_app_rx.recv::<AppResponse>().await.is_ok() {} });
+
+    // Alice's session should be in state unresolved.
+    // Leave some time for the countersigning workflow to attempt to resolve the session.
+    tokio::time::timeout(Duration::from_secs(20), async {
+        loop {
+            let state = get_session_state(&alice.cell_id, &alice_app_tx).await;
+            println!("state is {state:?}");
+            if matches!(
+                get_session_state(&alice.cell_id, &alice_app_tx).await,
+                Some(CountersigningSessionState::Unknown { resolution, .. }) if resolution.attempts >= 1
```

Comment:
@ThetaSinner The problem now is that resolving the session involves querying the authorities. Bob is the only other authority and Bob is shut down, so the resolution takes a long time (to fail). I guess the querying has to be time-boxed.

Reply:
For the test I can bring up Bob sooner, but I suggest to let the querying of the authorities time out as well in the workflow.

Reply:
Actually it's more random that the test passes like that. Since only the other signers (= Bob) are queried, the test doesn't make sense like that.